File: DojoArraySum_Golang/somador_test.go
package main

import (
	"testing"
)

func Test_SomarValoresArray(t *testing.T) {
	// Arrange
	esperado := 9
	vetorParaSomar := []int{2, 3, 4}

	// Act
	resultado := Somador(vetorParaSomar)

	// Assert
	if resultado != esperado {
		t.Error(
			"Esperado", esperado,
			"Obteve", resultado,
		)
	}
}

File: DojoArraySum_PHP/tests/SomadorTest.php
<?php

namespace App;

use PHPUnit\Framework\TestCase;
use App\Somador;

class SomadorTest extends TestCase
{
    public function testSomadorRetornaValorElementosSomados(): void
    {
        // Arrange
        $somador = new Somador();
        $esperado = 9;
        $vetorParaSomar = [2, 3, 4];

        // Act
        $result = $somador->somar($vetorParaSomar);

        // Assert
        $this->assertEquals($esperado, $result);
    }
}
